src/transaction_manager/transaction_manager.cpp

```diff
@@ -412,28 +350,71 @@ void TransactionManager::packTrxs(vec_trx_t &to_be_packed_trx, uint16_t max_trx_
                  [](Transaction const &t) { return t.getHash(); });
 }
 
+// Save the transaction that came with the block together with the
+// transactions that are in the queue. This will update the transaction
+// status as well and remove the transactions from the queue
 bool TransactionManager::verifyBlockTransactions(DagBlock const &blk, std::vector<Transaction> const &trxs) {
-  bool invalidTransaction = false;
+  vec_trx_t const &all_block_trx_hashes = blk.getTrxs();
+  if (all_block_trx_hashes.empty()) {
+    return true;
+  }
+
+  std::set<trx_hash_t> known_trx_hashes(all_block_trx_hashes.begin(), all_block_trx_hashes.end());
+
+  auto trx_batch = db_->createWriteBatch();
   for (auto const &trx : trxs) {
-    auto valid = verifyTransaction(trx);
-    if (!valid.first) {
-      invalidTransaction = true;
-      LOG(log_er_) << "Invalid transaction " << trx.getHash().toString() << " " << valid.second;
+    known_trx_hashes.erase(trx.getHash());
+    auto status = db_->getTransactionStatus(trx.getHash());
```

Comment:
Lets optimize this one. You already have vector of all txs hashes, there is a method to get all txs statuses from db at once and parameter is vector of hashes...

Reply:
hmm okay you dont have the vector of hashes, but might be still worth to loop over txs first and get hashes + 1 db call instead off N db calls.... ???

Reply:
but status can be outdated as verification of all unverified trxs is done in background by other threads, so this way as it is you will get as much up to date status as possible. And one more thing reading from DB is cheap, re-validating is expensive